### PR TITLE
Staging

### PR DESCRIPTION
**Pull request by mesudpasic**

### CODE REVIEW

File: Module.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
+version: 1.0.0
+name: Translator Module
```
Comment (kubagrzelak) [inline Module.yaml:2]: Please remove word "Module"

File: Module.yaml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+description: Module translates commands from application to device compatible commands
+icon: https://icons-020-demo.s3.eu-central-1.amazonaws.com/mi_translator_process.png
+type: process
+tags:
+  - Translator
+categories:
+  - id: category
+    name: Translator
```
Comment (kubagrzelak) [inline Module.yaml:10]: Please change category to `Policies`

File: Module.yaml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    value: ''
+    default: 0.0.0.0
+    type: string
+    name: Host Name
+  - key: INGRESS_PORT
+    description: Port on which the server will run.
+    value: ''
+    default: 80
+    type: integer
+    name: Port
```
Comment (kubagrzelak) [inline Module.yaml:26]: Please remove this.

File: README.md [hunk 4]
```diff
@@ -1,2 +1,90 @@
-# translator
-Module for translating commands from the App to device specific
+# Translator Module
```
Comment (kubagrzelak) [inline README.md:1]: Please remove word "Module"

File: README.md [hunk 5]
```diff
@@ -1,2 +1,90 @@
...
+
+|                |                             |
+| -------------- | --------------------------- |
+| Name           | Translator Module           |
```
Comment (kubagrzelak) [inline README.md:5]: Please remove word "Module"

File: README.md [hunk 6]
```diff
@@ -1,2 +1,90 @@
...
+| Version        | v1.0.0                      |
+| Dockerhub Link | [weevenetwork/translator]() |
+| Authors        | Mesud Pasic                 |
+
+- [MQTT Ingress](#translator-module)
```
Comment (kubagrzelak) [inline README.md:10]: Please change MQTT Ingress -> Translator and remove word "Module" from the brackets

File: README.md [hunk 7]
```diff
@@ -1,2 +1,90 @@
...
+  - [Description](#description)
+  - [Features](#features)
+  - [Environment Variables](#environment-variables)
+    - [Module Specific](#module-specific)
+    - [Set by the weeve Agent on the edge-node](#set-by-the-weeve-agent-on-the-edge-node)
+  - [Dependencies](#dependencies)
+
+## Description
+
+Encoder for Vicki LoRaWAN's commands.
```
Comment (kubagrzelak) [inline README.md:20]: Please be more descriptive and informative. I have no idea what this module does by reading this sentence.

File: README.md [hunk 8]
```diff
@@ -1,2 +1,90 @@
...
+## Features
+
+- Parsing Melita.io data for thermostat
+- Sends data to next module service via REST API
+
+## Environment Variables
+
+- INGRESS_HOST
+- INGRESS_PORT
+- DEVICE_MANUFACTURER
```
Comment (kubagrzelak) [inline README.md:31]: By Environmental Variables we mean the variables that are set by a user in the UI (so the ones from YAML file). Please construct a table similar to the tables in the same section of McClimate Vicki Encoder: https://github.com/weeve-modules/mcclimate-vicki-encoder/blob/main/README.md

File: README.md [hunk 9]
```diff
@@ -1,2 +1,90 @@
...
+### Module Specific
+
+### Set by the weeve Agent on the edge-node
+
+| Environment Variables | type | Description |
+| --- | --- | --- |
+| MODULE_NAME | string | Name of the module |
+| INGRESS_HOST | string | Host where app is running |
+| INGRESS_PORT | string | Port where app is running |
+| DEVICE_MANUFACTURER | string | Name of the device manufacturer that will map to JSON file where mapping is read for commands and parameters |
```
Comment (kubagrzelak) [inline README.md:42]: Please remove this entry, as it should be mentioned above in Environmental Variables table as per my previous comment.

File: README.md [hunk 10]
```diff
@@ -1,2 +1,90 @@
...
+
+- Module translates JSON command input from MongoDB (application) to compatible command interface for specific device, so that encoder module can convert it to valid command
```
Comment (kubagrzelak) [inline README.md:44]: This is a good sentence that should be moved to Description section

File: docker-compose.yml [hunk 11]
```diff
@@ -0,0 +1,10 @@
+version: '1'
+services:
+  translator_service:
+    build:
+      context: .
+      dockerfile: Dockerfile
+    environment:
+      MODULE_NAME: translator_service
+      INGRESS_HOST: 127.0.0.1
+      INGRESS_PORT: 80
```
Comment (kubagrzelak) [inline docker-compose.yml:10]: Please add all environmental variables that are set in the UI (the ones from the YAML file) and provide their examples, so someone calling docker compose command could run this file.

File: docker-compose.yml [hunk 12]
```diff
@@ -0,0 +1,10 @@
+version: '1'
+services:
+  translator_service:
```
Comment (kubagrzelak) [inline docker-compose.yml:3]: please change translator_service to translator (name of the module)

File: docker-compose.yml [hunk 13]
```diff
@@ -0,0 +1,10 @@
...
+    build:
+      context: .
+      dockerfile: Dockerfile
+    environment:
+      MODULE_NAME: translator_service
```
Comment (kubagrzelak) [inline docker-compose.yml:8]: please change translator_service to translator (name of the module)

File: image/src/config/config.js [hunk 14]
```diff
@@ -0,0 +1,9 @@
+const env = require('../utils/env')
+
+module.exports = {
+  INGRESS_HOST: env('INGRESS_HOST', '127.0.0.1'),
+  INGRESS_PORT: env('INGRESS_PORT', '8080'),
+  MODULE_NAME: env('MODULE_NAME', 'Translator Module'),
```
Comment (kubagrzelak) [inline image/src/config/config.js:6]: Please change MODULE_NAME to `translator`

File: image/src/utils/util.js [hunk 15]
```diff
@@ -0,0 +1,15 @@
+const { NESTED_RESPONSE, CUSTOM_FIELDS } = require('../config/config')
```
Comment (kubagrzelak) [inline image/src/utils/util.js:1]: If we don't use NESTED_RESPONSE or CUSTOME_FIELDS then please remove this line

**Review by kubagrzelak — CHANGES_REQUESTED** [review]

File: Module.yaml (hunk 1 above)
Comment (mesudpasic) [reply Module.yaml:2]: done

File: Module.yaml (hunk 2 above)
Comment (mesudpasic) [reply Module.yaml:10]: done

File: Module.yaml (hunk 3 above)
Comment (mesudpasic) [reply Module.yaml:26]: done

File: README.md (hunk 4 above)
Comment (mesudpasic) [reply README.md:1]: done

File: README.md (hunk 5 above)
Comment (mesudpasic) [reply README.md:5]: done

File: README.md (hunk 6 above)
Comment (mesudpasic) [reply README.md:10]: done

File: README.md (hunk 7 above)
Comment (mesudpasic) [reply README.md:20]: done.

File: README.md (hunk 8 above)
Comment (mesudpasic) [reply README.md:31]: done.

File: README.md (hunk 10 above)
Comment (mesudpasic) [reply README.md:44]: done

File: README.md (hunk 9 above)
Comment (mesudpasic) [reply README.md:42]: done

File: docker-compose.yml (hunk 11 above)
Comment (mesudpasic) [reply docker-compose.yml:10]: added DEVICE_MANUFACTURER

File: docker-compose.yml (hunk 12 above)
Comment (mesudpasic) [reply docker-compose.yml:3]: done

File: docker-compose.yml (hunk 13 above)
Comment (mesudpasic) [reply docker-compose.yml:8]: done

File: image/src/config/config.js (hunk 14 above)
Comment (mesudpasic) [reply image/src/config/config.js:6]: done

File: image/src/utils/util.js (hunk 15 above)
Comment (mesudpasic) [reply image/src/utils/util.js:1]: removed

**Review by kubagrzelak — APPROVED** [review]

**Pull request merged**